providers/gcs/gcs.go

```diff
@@ -42,6 +43,10 @@ type Config struct {
 	Bucket         string `yaml:"bucket"`
 	ServiceAccount string `yaml:"service_account"`
 	UseGRPC        bool   `yaml:"use_grpc"`
+	// Both upload and download paths will use zonal bucket gRPC APIs by default.
+	// Zonal buckets are currently allowlisted; please contact your Google account
+	// manager if interested.
```

Comment:
```suggestion
	// UseZonalBuckets enables Zonal gRPC APIs to be used to access (both writes and reads) GCS storage. You can read more about zonal buckets <here>
	//
	// Initial benchmarks shown avg. ~50+% reduction of latency for larger Thanos store operations (see: https://github.com/thanos-io/objstore/pull/209#issue-3294127783).
```

Reply:
I would leave the following info in some public doc and link it from here. Otherwise this comment might age very fast.

> Zonal buckets are currently in Google Private Preview; please contact your Google account
>  manager if interested.

Reply:
As written in suggestion, I would also ignore UseGRPC if zonal is enabled - might be easier for user - not a strong opinion though.